pkg/openstack/client/networking.go

```diff
@@ -286,8 +286,18 @@ func (c *NetworkingClient) GetRouterInterfacePort(routerID, subnetID string) (*p
 	if err != nil {
 		return nil, err
 	}
-	if len(list) == 0 {
+
+	validDeviceOwners := []string{
+		"network:router_interface",
+		"network:router_interface_distributed",
+		"network:ha_router_replicated_interface",
+	}
+	filtered := slices.DeleteFunc(list, func(p ports.Port) bool {
+		return !slices.Contains(validDeviceOwners, p.DeviceOwner)
+	})
```

Comment:
lgtm

Out of curiosity, I have to ask why did you go for the delete compared to something like
```
	for _, v := range list {
		if slices.Contains(validDeviceOwners, v.DeviceOwner) {
			return &v, nil
		}
	}
	return nil, nil
```

Reply:
tbh in the moment when fixing this I found it to be more elegant. I'm not too familiar with go yet and I was searching the `slices` package for something like `filter` in python or `select` in ruby.